core/src/Kokkos_Parallel_Reduce.hpp

```diff
@@ -99,6 +99,10 @@ struct Sum {
   bool references_scalar() const { return references_scalar_v; }
 };
 
+template <typename Scalar, typename... Properties>
+Sum(View<Scalar, Properties...> const&)
+    ->Sum<Scalar, typename View<Scalar, Properties...>::memory_space>;
+
```

Comment:
Should we have this:
```c++
template<typename Scalar>
Sum(Scalar&) -> Sum<Scalar, AnonymousSpace>;
```
and would we need some sfinae/requires stuff on it to make it not trigger with Views?

Reply:
Thought that AnonymousSpace would be problematic, since it reads as "device accessible". We would need to switch our logic in the reduction where we check that, to make AnonymousSpace be treated as not not-device accessible maybe?